### DISCUSSION TITLE
Installation of parallel MEEP with openmp without root access

### DISCUSSION THREAD

**Discussion by olibasquette** [NanoComp/meep#2635] [General]
I've been using parallel MEEP for the last few weeks to model far field reflections from certain 3D structures of interest. This is getting rather slow as the structures I create get more complicated. I have tried to build MEEP from source on my group's server with openmp support to hopefully speed things up, as suggested in the docs. Unfortunately I don't have root access, so the majority of the scripts given at https://meep.readthedocs.io/en/latest/Build_From_Source/#building-from-source don't work.

I was wondering if anyone here has had experience with something similar and how they got around it? Alternatively, if there are any pre-compiled conda files available that would be really useful.

Thanks!

**Comment by smartalecH** [comment]
If the bottleneck is computing a Near2Far transform, then OpenMP will help. But OpenMP isn't going to speed up the main timestepping. Currently there is too much cache contention with the current implementation, and MPI will consistently give better performance. 

I would keep this in mind before going through the hassle of rebuilding everything.

**Reply by olibasquette** [reply]
Thanks for your comment. I believe the speed of the main timestepping is okay at the moment (when using MPI) so yes the bottleneck is the Near2Far transform.
